examples/rpc/partner-plugin/handlers/accommodation/v1/accommodation_search.go

```diff
@@ -0,0 +1,271 @@
+	md := metadata.Metadata{}
+
+	var search_generic_params = req.SearchParametersGeneric
+	// print params
+	fmt.Printf("Search generic params: %+v\n", search_generic_params)
+
+	if err := md.ExtractMetadata(ctx); err != nil {
+		log.Print("error extracting metadata")
+	}
+
+	md.Stamp(fmt.Sprintf("%s-%s", "ext-system", "response"))
+
+	log.Printf("Responding to request (Accommodation Search): %s", md.RequestID)
+
+	// load mock data
+	properties := helpers.LoadPropertiesMockData()
+
+	// log
+	fmt.Printf("properties: %+v\n", properties)
+
+	// if there is no query, return no results
+	if len(req.Queries) == 0 {
+		return &accommodationv1.AccommodationSearchResponse{
+			Header: &typesv1.ResponseHeader{
+				Status: typesv1.StatusType_STATUS_TYPE_FAILURE,
+				Alerts: []*typesv1.Alert{
+					{
+						Message: "No queries provided",
+						Type:    typesv1.AlertType_ALERT_TYPE_INFO,
+					},
+				},
+			},
+		}, nil
+	}
+
+	searchResults := []*accommodationv1.AccommodationSearchResult{}
+	available_properties := []*accommodationv1.PropertyExtendedInfo{}
+	// loop request queries
+	for _, query := range req.Queries {
+		props := make([]*accommodationv1.PropertyExtendedInfo, len(properties))
+		for i := range properties {
+			props[i] = &properties[i]
+		}
+
+		// get filtered properties
+		var filtered_props = filterPropertiesByGeoTreeLocation(props, query.SearchParametersAccommodation.GetLocationGeoTree())
+		// filter by product codes
+		filtered_props = filterPropertiesByProductCodes(filtered_props, query.SearchParametersAccommodation.GetProductCodes())
+		// filter by supplier codes
+		filtered_props = filterPropertiesBySupplierCodes(filtered_props, query.SearchParametersAccommodation.GetSupplierCodes())
+
+		// loop filtered properties and check if they are already in available_properties
+		for _, prop := range filtered_props {
+			// Check if property already exists in available_properties
+			exists := false
+			for _, existingProp := range available_properties {
+				if existingProp.Property.SupplierCode.SupplierCode == prop.Property.SupplierCode.SupplierCode {
+					exists = true
+					break
+				}
+			}
+			if !exists {
+				available_properties = append(available_properties, prop)
+			}
+		}
+
+		var price = 2
+
+		// generate search result
+		for _, prop := range available_properties {
+
+			// units requested
+			units_requested := query.UnitCount
+
+			// empty units array
+			units := make([]*accommodationv1.Unit, 0)
+
+			// loop all rooms
+			for _, room := range prop.Rooms {
+
+				units = append(units, &accommodationv1.Unit{
+					Type:             0,
+					SupplierRoomCode: room.SupplierCode,
+					SupplierRoomName: room.SupplierName,
+					OriginalRoomName: room.OriginalName,
+					TravelPeriod: &typesv1.TravelPeriod{
+						StartDate: &typesv1.Date{
+							Year:  query.TravelPeriod.GetStartDate().GetYear(),
```

Comment:
Filtering for a travel period gets a bit tricky if the dates are generated from the query. I'm not sure if that is a good idea. 
Is there a reason why the mock data doesn't just define static dates which are documented and always valid (even if they're in the past)?

Reply:
Added a period where valid results will be returned -> 01.06.2025 - 30.06.2025